### PR TITLE
feat(@angular/cli): support TypeScript 2.4

### PR DESCRIPTION
**Pull request by filipesilva**
Fix https://github.com/angular/angular-cli/issues/6827

### CODE REVIEW

**Comment by filipesilva** [comment]
This PR is up mostly to experiment. It's not meant to go in until Angular itself officially support TS 2.4.

File: packages/@angular/cli/commands/new.ts [hunk 1]
```diff
@@ -176,8 +176,8 @@ const NewCommand = Command.extend({
           }
         });
     } else {
-      createDirectory = mkdir(directoryName)
-        .catch(err => {
+      createDirectory = (mkdir as any)(directoryName)
```
Comment (hansl) [inline packages/@angular/cli/commands/new.ts]: `mkdir` should be the right type, or the type of `denodeify` itself is wrong.

**Comment by fxck** [comment]
Any idea when will angular update to 2.4.x?

File: packages/@angular/cli/commands/new.ts (hunk 1 above)
Comment (filipesilva) [inline packages/@angular/cli/commands/new.ts]: It seems that something is wrong with either `@types/node` or `@types/denodeify`. I Updated the versions but the error was still there. Added note to explain why it is being typed as `any` and a TODO for me to fix.

**Comment by filipesilva** [comment]
New pr: https://github.com/angular/angular-cli/pull/7065

**Pull request closed without merging**